src/components/dom.js

```diff
@@ -86,6 +98,55 @@ export default class Dom {
 
     }
 
+    /**
+     * Search for deepest node
+     *
+     * @param {Element} node - start Node
+     * @param {Boolean} atLast - find last text node
+     * @return {*}
+     */
+    static getDeepestTextNode(node, atLast = false) {
+
+        if (node.childNodes.length === 0) {
+
+            /**
+             * We need to return empty text node
+             * But caret will not be placed in empty textNode, so we need textNode with zero-width char
```

Comment:
Maybe this part doesn't apply to this method? We can make a decorator around `getDeepestTextNode`: something  like `getNotEmptyDeepestTextNode`, when will be placed checking for an emptiness and adding a zero-width character